UnitTests/BraintreeTestShared/Encodable+Extension.swift

```diff
@@ -10,14 +9,18 @@ public extension Encodable {
         do {
             let data = try encoder.encode(self)
             let object = try JSONSerialization.jsonObject(with: data)
-            
+
             if let json = object as? [String: Any] {
                 return json
             } else {
-                throw BTHTTPError.serializationError("Serialization to dictionary failed.")
+                throw NSError(
```

Comment:
qq here - is there a reason why we choose to throw a custom `NSError` type versus the original `BTHTTPError.serializationError()`